medallion/backends/memory_backend.py

```diff
@@ -155,25 +156,28 @@ def add_objects(self, api_root, id_, objs, request_time):
                 if "id" in collection and id_ == collection["id"]:
                     if "objects" not in collection:
                         collection["objects"] = []
-                    for new_obj in objs["objects"]:
-                        id_and_version_already_present = False
-                        if new_obj["id"] in collection["objects"]:
-                            current_obj = collection["objects"][new_obj["id"]]
-                            if "modified" in new_obj:
-                                if new_obj["modified"] == current_obj["modified"]:
+                    try:
+                        for new_obj in objs["objects"]:
+                            id_and_version_already_present = False
+                            if new_obj["id"] in collection["objects"]:
+                                current_obj = collection["objects"][new_obj["id"]]
+                                if "modified" in new_obj:
+                                    if new_obj["modified"] == current_obj["modified"]:
+                                        id_and_version_already_present = True
+                                else:
+                                    # There is no modified field, so this object is immutable
                                     id_and_version_already_present = True
+                            if not id_and_version_already_present:
+                                collection["objects"].append(new_obj)
+                                self._update_manifest(new_obj, api_root, collection["id"])
+                                successes.append(new_obj["id"])
+                                succeeded += 1
                             else:
-                                # There is no modified field, so this object is immutable
-                                id_and_version_already_present = True
-                        if not id_and_version_already_present:
-                            collection["objects"].append(new_obj)
-                            self._update_manifest(new_obj, api_root, collection["id"])
-                            successes.append(new_obj["id"])
-                            succeeded += 1
-                        else:
-                            failures.append({"id": new_obj["id"],
-                                             "message": "Unable to process object"})
-                            failed += 1
+                                failures.append({"id": new_obj["id"],
+                                                 "message": "Unable to process object"})
+                                failed += 1
+                    except Exception as e:
```

Comment:
What types of errors typically occur here? I would prefer a smaller `try` block and some more specific exceptions being caught, if it's possible to do that. Otherwise this is a pretty general "something went wrong" error handler. If we can give more information to the user in the response about *why* the request couldn't be processed (without going overboard), I think that would help.

Reply:
Lol great question! The idea is that it surrounds a code block of access/manipulations to user supplied STIX where if the STIX content is malformed an error will occur, e.g. keyerror, Nonetype, missing attribute etc... I didnt trust my ability to map out all possible errors that could arise thus I left the (much frowned upon) general try/except.

So the user will be supplied the underlying exception (e.g. keyerror, nonetype) but there is no context (e.g. stacktrace) with it. I didnt want to encode possibly more specific static error messages as they could be completely off for an unforeseen error.